lib/mongoid/association/referenced/has_many/proxy.rb

```diff
@@ -217,9 +217,18 @@ def each(&block)
           # @example Are there persisted documents?
           #   person.posts.exists?
           #
+          # @param [ :none | nil | false | Hash | Object ] id_or_conditions
+          #   When :none (the default), returns true if any persisted
+          #   documents exist in the association. When nil or false, this
+          #   will always return false. When a Hash is given, this queries
+          #   the documents in the association for those that match the given
+          #   conditions, and returns true if any match. Any other argument is
+          #   interpreted as an id, and queries for the existence of documents
+          #   in the association with a matching _id.
+          #
           # @return [ true | false ] True is persisted documents exist, false if not.
-          def exists?
-            criteria.exists?
+          def exists?(id_or_conditions = :none)
```

Comment:
Why not:

```
delegate :exists?, to: :criteria
```

Reply:
That would work, too. In general I'm not a fan of that style. It tends to stymie investigation into execution flow, at least for me. I'll think about it, though.

Reply:
It is recommended by Rubocop: https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsdelegate